go/kbfs/search/indexer_test.go

```diff
@@ -0,0 +1,195 @@
+// Copyright 2020 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package search
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/blevesearch/bleve"
+	"github.com/keybase/client/go/kbfs/data"
+	"github.com/keybase/client/go/kbfs/idutil"
+	"github.com/keybase/client/go/kbfs/libcontext"
+	"github.com/keybase/client/go/kbfs/libkbfs"
+	"github.com/keybase/client/go/kbfs/tlf"
+	"github.com/keybase/client/go/kbfs/tlfhandle"
+	"github.com/keybase/client/go/logger"
+	"github.com/stretchr/testify/require"
+)
+
+func testInitConfig(
+	ctx context.Context, config libkbfs.Config, session idutil.SessionInfo,
+	log logger.Logger) (
+	newCtx context.Context, newConfig libkbfs.Config,
+	configShutdown func(context.Context) error, err error) {
+	configLocal, ok := config.(*libkbfs.ConfigLocal)
+	if !ok {
+		panic(fmt.Sprintf("Wrong config type: %T", config))
+	}
+
+	newConfig = libkbfs.ConfigAsUserWithMode(
+		configLocal, session.Name, libkbfs.InitSingleOp)
+
+	kbCtx := config.KbContext()
+	params, err := Params(kbCtx, config.StorageRoot(), session.UID)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	newConfig.(*libkbfs.ConfigLocal).SetStorageRoot(params.StorageRoot)
+
+	mdserver, err := libkbfs.MakeDiskMDServer(config, params.StorageRoot)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	newConfig.SetMDServer(mdserver)
+
+	bserver := libkbfs.MakeDiskBlockServer(config, params.StorageRoot)
```

Comment:
Is there a reason we use disk servers rather than the memory one?

Reply:
I spent some time just now trying to use a memory-based one, but it turns out it doesn't work.  The reason is that bleve itself writes a config file to the local `libfs.FS` you pass it, which stores the kvstore config.  This is only done once, at start up (which happens before the user login notification).  Once login happens, the indexer is torn down and re-built up using the new user's private TLF to encrypt stuff; the kvstore isn't re-registered, so if you're using a memory-based server, that original config file (which is device-specific, not user specific) has been lost, and the indexer just blows up and dies.

I'll add a comment there about why we're using disk-based stuff in the test.

Reply:
Ah, got you. That makes sense. Thanks for writing that up!